modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperCppEngine.java

```diff
@@ -0,0 +1,417 @@
+      logger.debug("WhisperC++ log prob threshold set to {}", whispercppLogProbThreshold);
+    }
+
+    whispercppDiarization = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_DIARIZATION_CONFIG_KEY);
+    if (whispercppDiarization.isSome()) {
+      logger.debug("WhisperC++ diarization set to {}", whispercppDiarization);
+    }
+
+    whispercppTinyDiarization = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_TINY_DIARIZATION_CONFIG_KEY);
+    if (whispercppTinyDiarization.isSome()) {
+      logger.debug("WhisperC++ tiny diarization set to {}", whispercppTinyDiarization);
+    }
+
+    whispercppNoFallback = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_NO_FALLBACK_CONFIG_KEY);
+    if (whispercppNoFallback.isSome()) {
+      logger.debug("WhisperC++ no fallback set to {}", whispercppNoFallback);
+    }
+
+    logger.debug("Finished activating/updating speech-to-text service");
+  }
+
+  //TODO: Add method for language detection
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.speechtotext.api.SpeechToTextEngine#generateSubtitlesFile(File, File, String, Boolean)
+   */
+
+  @Override
+  public Map<String, Object> generateSubtitlesFile(File mediaFile,
+      File preparedOutputFile, String language, Boolean translate)
+          throws SpeechToTextEngineException {
+
+    if (!FilenameUtils.isExtension(mediaFile.getPath(), "wav")) {
+      throw new SpeechToTextEngineException("WhisperC++ currently doesn't support any media extension other than wav");
+    }
+
+    String[] baseCommands = { whispercppExecutable,
+        mediaFile.getAbsolutePath(),
+        "--model", whispercppModel,
+        "-ovtt",
+        "-oj",
+        "--output-file", preparedOutputFile.getAbsolutePath().replaceFirst("[.][^.]+$", "")};
+
+    List<String> command = new ArrayList<>(Arrays.asList(baseCommands));
+
+    if (whispercppBeamSize.isSome()) {
+      command.add("-bs");
+      command.add(Integer.toString(whispercppBeamSize.get()));
+    }
+    if (whispercppMaxLength.isSome()) {
+      command.add("-ml");
+      command.add(Integer.toString(whispercppMaxLength.get()));
+    }
+    if (whispercppThreads.isSome()) {
+      command.add("-t");
+      command.add(Integer.toString(whispercppThreads.get()));
+    }
+    if (whispercppProcessors.isSome()) {
+      command.add("-p");
+      command.add(Integer.toString(whispercppProcessors.get()));
+    }
+    if (whispercppMaxContext.isSome()) {
+      command.add("-mc");
+      command.add(Integer.toString(whispercppMaxContext.get()));
+    }
+    if (whispercppSplitOnWord.isSome() && whispercppSplitOnWord.get()) {
+      command.add("-sow");
+    }
+    if (whispercppBestOf.isSome()) {
+      command.add("-bo");
+      command.add(Integer.toString(whispercppBestOf.get()));
+    }
+    if (whispercppWordThreshold.isSome()) {
+      command.add("-wt");
+      command.add(String.format("%f", whispercppWordThreshold.get()));
```

Comment:
This depends on the system's language settings and may break.
If the system language is English, this will result in something like `1.1`:

```
❯ echo $LANG
en_US.UTF-8

❯ jshell    
jshell> String.format("%f", 1.1)
$1 ==> "1.100000"
```

But switching to German, for example, this will be `1,1` which I assume will cause a problem?
```
❯ export LANG=de_DE.UTF-8

❯ jshell                 
jshell> String.format("%f", 1.1)
$1 ==> "1,100000"
```

This is used at more places down below.

You are also converting the configuration values to numbers, only to convert them back here. You could just never convert them.

Reply:
You're right. Converting the configuration values to numbers, although somewhat unnecessary, is useful to prevent unwanted behavior when incompatible config values are passed to whispercpp.  This way it is ensured that whispercpp will behave as expected. I will fix the format issue.